### ISSUE TITLE
pcap-hurd.c: cppcheck warning "When using void pointers in calculations, the behaviour is undefined"

### ISSUE THREAD

**Issue by fxlb** [the-tcpdump-group/libpcap#1173]
```
pcap-hurd.c:77:18: warning: 'p->buffer' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined. [arithOperationsOnVoidPointer]
 pkt = p->buffer + offsetof(struct net_rcv_msg, packet)
                 ^
pcap-hurd.c:78:8: warning: 'p->buffer+offsetof(struct net_rcv_msg,packet)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined. [arithOperationsOnVoidPointer]
       + sizeof(struct packet_header) - ETH_HLEN;
       ^
pcap-hurd.c:79:25: warning: 'p->buffer' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined. [arithOperationsOnVoidPointer]
 memmove(pkt, p->buffer + offsetof(struct net_rcv_msg, header),
```

**Comment by infrastation** [comment]
`pcap_read_nocb_remote()` and `usb_read_linux_bin()` just cast the pointer to `u_char *` in a similar context, so supposedly that works, alignment permitting.

**Comment by guyharris** [comment]
And if alignment *doesn't* permit. then 1) the Hurd developers should have made it permit and 2) that would ahve to be fixed with copying if the Hurd supports any machines that don't fully support unaligned accesses (or maybe support it by trapping to code that simulates the operation, which generally means "simulates the operation *slowly*").

So casting is probably the right thing to do here.

**Comment by fxlb** [comment]
Why not a change from `void *buffer` to `u_char *buffer` in `struct pcap` ?

**Comment by infrastation** [comment]
I was going to write "because `struct pcap` is in the public part of libpcap API", but it is in fact in `pcap-int.h`, so maybe that would be practicable.

**Comment by fxlb** [comment]
I have added the PR #1183 to check this update with all CI systems.

**Comment by infrastation** [comment]
@guyharris, does a change from `void *` to `u_char *` count as an ABI change?

**Comment by infrastation** [comment]
Thank you. So everything should be alright then (famous last words, eh?).

**Comment by fxlb** [comment]
Fixed by f8e510aaefcb0c229ce72116c3b6902bf2e3da63 via PR #1183.
